Added many new buttons

blackshadowshade opened the pull request:
Addresses many of the elements in #2535.

I have no idea whether copyright has been cleared and permissions obtained for all these buttons, I'd like those confirmed before we accept this pull request.

I'm also unsure whether there is flavour text that needs to be added for these buttons.

[dev site: https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com ]


To do:
- [x] Update flavour text for Parasprites and Rarity
- [x] Add remaining Button Ponies images when they are supplied

TheOrgg commented:
Everything has been cleared except the Scott Pilgrim artwork, but that's from the book covers.  That's about as cleared as it can get for a mass media item, so it's about as clear as the Cowboy Bebop buttons.

Everything else we've got permission for other than Space Battles and Twin Sword Training, I believe.

TheOrgg commented:
Scrolled through the images.  I did Weaver friendly versions (150px w/ shadow) for the Standard Action buttons.  They're on the linked issue.

TheOrgg commented:
Would you like shadow/shine on the Scott Pilgrim buttons?

TheOrgg commented:
I think I'd prefer the squared off version of the Julz button, but that's just my preference.  It seems a little clearer.  If ya'll want the button picture I can crop it a bit more.

blackshadowshade commented:
> Scrolled through the images. I did Weaver friendly versions (150px w/ shadow) for the Standard Action buttons. They're on the linked issue.

Those two images have been updated

blackshadowshade commented:
> I think I'd prefer the squared off version of the Julz button, but that's just my preference. It seems a little clearer. If ya'll want the button picture I can crop it a bit more.

Actually, if we shrink this to the standard size, that may address the issue.

irilyth commented:
The Julz art on this PR seems weird to me -- there's like a ring of some sort around the button itself?

blackshadowshade commented:
> The Julz art on this PR seems weird to me -- there's like a ring of some sort around the button itself?

That looks to me like an actual metal button that has been photographed.

TheOrgg commented:
> The Julz art on this PR seems weird to me -- there's like a ring of some sort around the button itself?

This is an actual picture from a website.  I still am pulling for the 150px square Julz.  I think it's kinda neat!

blackshadowshade commented:
> > The Julz art on this PR seems weird to me -- there's like a ring of some sort around the button itself?
> 
> This is an actual picture from a website. I still am pulling for the 150px square Julz. I think it's kinda neat!

For those who don't know what TheOrgg is talking about, the square image is this one: 
![](https://user-images.githubusercontent.com/44233993/71772961-69dd7e00-2f1a-11ea-8caa-33a7f0bce754.png)

I personally think that this one looks terrible compared to the round one (which we could crop if necessary).

blackshadowshade commented:
I'll note that TheOrgg has mentioned potentially leaving out the three Button Pony buttons with duplicate recipes (the clones of Changeling, Lab Rat, and Bunnies). I would also be fine with this.

TheOrgg commented:
![150pxjulz](https://github.com/user-attachments/assets/2da428a4-3ecc-40c1-bcd0-107a392358c8)
Better?

And how about K29's Enjelion?

blackshadowshade commented:
Yes, that image of Julz is much better!

And I have no idea what you're referring to when you talk about K29's Enjelion.

blackshadowshade commented:
Image for Julz has been updated.

dwvanstone commented:
> I'll note that TheOrgg has mentioned potentially leaving out the three Button Pony buttons with duplicate recipes (the clones of Changeling, Lab Rat, and Bunnies). I would also be fine with this.

I actually think it's fine if we add Changeling, Angel, and Parasprites if they are important characters in the Button Pony universe. It doesn't matter that their recipes match -- it's the fun of getting to play your favorite buttons from a fan set. (In my opinion.)

TheOrgg commented:
> And I have no idea what you're referring to when you talk about K29's Enjelion.

![](https://user-images.githubusercontent.com/44233993/105919032-8dc6a280-5ffa-11eb-9c9a-e7a557ee89d6.png)
https://buttonmen.fandom.com/wiki/Cryptonom

blackshadowshade commented:
Enjileon added

blackshadowshade commented:
Now begins the difficult bit: automated and manual testing of the new buttons.

TheOrgg commented:
How's the automatic test going?

cgolubi1 commented:
> How's the automatic test going?

It's on my list.  The database encoding thing is time-sensitive --- i spent about six hours yesterday root-causing over on #2974.  But we have a plan for that now, so hopefully i'll be unblocked from standing up some replay sites for PRs like this soon.

TheOrgg commented:
*nudge nudge*, question mark?

cgolubi1 commented:
@TheOrgg (and others) dev site has been deployed to https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com --- check it out.

TheOrgg commented:
The Ponies set hasn't got artwork (but I know there's art on the Pony-specific issue).

TheOrgg commented:
I know the Ponies set has art, as does Wumpus.

blackshadowshade commented:
I've added the seven Button Ponies images that I have with shadow, plus the flavour texts that I have from above.

Two things grabbed my attention with the flavour text: Parasprites being inconsistently in the past tense in the second sentence; and Rarity having a comma in the first sentence that feels wrong to me.

blackshadowshade commented:
(Also remember, my changes won't be reflected on the testing site until after the site has been explicitly updated.)

cgolubi1 commented:
It looks like people have started testing on the dev site, so i won't rebuild it right now.

cgolubi1 commented:
I stood up a replay site (also using v1, which should be fine) and am starting testing:

```
./replay_loop -pb 'Julz,Eberk,Bunny Temple,Bernie,GG,Wumpus,Edda,Fernando,Applejack,Chrysalis,Fluttershy,Pinkie Pie,Rainbow Dash,Rarity,Twilight Sparkle,Changeling (BP),Angel (BP),Parasprites,Envy Adams,Final Form GGG,Gideon G Graves,Lucas Lee,Matthew Patel,Nega Scott,Ramona Flowers,Scott Pilgrim,Roxy Richter,Todd Ingram,Katayanagi Twins,Knives Chau,Enjileon' -l
```

TheOrgg commented:
I agree with the comma being unneeded.  However, the Parasprites were not in ponyville for very long, so when they were in ponyville, that's what they enjoyed.  I think their text is correct.

And as said on the Pony thread, I'll work on art as soon as I can.  New computer, plus no time.

blackshadowshade commented:
For the Parasprites flavour text, maybe we could revise the second sentence to starting with

"While they were in Ponyville, they ..."

TheOrgg commented:
That should work.

blackshadowshade commented:
Sure, not a problem. I'll do that when I have the additional Button Ponies images.

cgolubi1 commented:
Actually looking at these button recipes in terms of what issues we expect to see:
* Fluttershy and Twilight Sparkle have unimplemented skills, so i don't expect them to be available on the prod site (which is fine).
* Enjileon has a moderately complex recipe (just, lots of twin dice with other skills, an area where we've historically had issues).
* I don't see anything else; let me know if i've missed something obvious.

The test site has played about 1000 novel games so far, with no issues.  (Replay testing is a bit munged up because of both the changes i've been making to RandomAI and the "invalid pass" backend fix not being merged in yet, but i'm not that worried about it; i don't expect replay testing to be interesting when all we're doing is adding new buttons.)

I've switched the site over to playing games using only Enjileon.  If that doesn't turn anything up, i'm not too concerned about this PR from a logic point of view.  I'll keep y'all posted.

cgolubi1 commented:
The site has played and replayed 600 Enjileon games since this morning with no errors.

I'm satisfied with automated testing of these buttons.

cgolubi1 commented:
@blackshadowshade when you get a moment, can you summarize what's outstanding on this PR?  I'd be in favor of putting a checklist in the description so we can keep track.

TheOrgg commented:
The art for Julz is the actual artwork from the button.  There's a whole set of "Cheapass Employees" that are written like that; Julz is the only one that was still available online. 

It's also the only one that's unique; the others have recipes that are identical to other CAG released buttons.

blackshadowshade commented:
I've added a to-do list for this issue.

cgolubi1 commented:
Tearing down the dev site as it hasn't seen any activity in 3 weeks.  I'll stand up a new one when there's a new revision to test.

TheOrgg commented:
With the additions on the original Pony page, is this one ready to go at some point?

cgolubi1 commented:
Blackshadowshade, as far as i know, this is on your plate with the changes TheOrgg posted to #2495, to put up a new revision following the checklist on top of this issue.  Let us know if that's not right, and there's anything anyone else needs to do.

(I don't think it's a tearing rush, just clarifying the status.  Thanks, all!)

blackshadowshade commented:
That's correct, this is for me to complete. I'll give it a go when I have a spare hour, it's a bit tough at the moment since it's both exam season and concert season in our household, plus I'm still trying to finish up at work.

TheOrgg commented:
> Absolutely no rush or need to explain - this is a hobby for all of us, and we very much appreciate everything you do!

What she said.  :)

blackshadowshade commented:
I think I've made the necessary changes and rebased. Testers, after this is deployed, take a good look at the new buttons, especially the Button Ponies images and their correspondence to the recipes.

cgolubi1 commented:
Okay, the dev site is back up at https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com/

Shadowshade's request from the last comment:

> Testers, after this is deployed, take a good look at the new buttons, especially the Button Ponies images and their correspondence to the recipes.

dwvanstone commented:
Should Bunny Temple have an image? I see it in #2535 but not on the dev site. TheOrgg: did you want to add description text since Bunny was someone you knew? It would make more sense on the site in the "Tribute" section to set the tone that this is where friends/family could be remembered.

Applejack's description should have "Earth" capitalized (see https://mlp.fandom.com/wiki/Applejack).
Same with Pinky Pie. (https://mlp.fandom.com/wiki/Pinkie_Pie)

Applejack's pet dog is spelled "Winona" not "Winnona".  (see https://mlp.fandom.com/wiki/Winona)

Should Rainbow Dash be listed as a "Pegasus pony" rather than a "pegasus"? (https://mlp.fandom.com/wiki/Rainbow_Dash) Or even just a "Pegasus"? 

Should Wumpus have an image? I see one in #2535.

Neat to see some new buttons!

TheOrgg commented:
Kit said the corrections to the flavor text were correct.  (and there were images of Wumpus and Bunny Temple, too.)

bobhope2 commented:
> Kit said the corrections to the flavor text were correct. (and there were images of Wumpus and Bunny Temple, too.)

Thank you.

blackshadowshade commented:
I've updated the flavour texts and added images according to @dwvanstone's last post. Pull request has also been rebased.

cgolubi1 commented:
Okay!  The dev site should be back up at https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com/ui/ with the new code.  Take a look, and if someone says this is good to go, i'll merge it.

dwvanstone commented:
Pinky Pie's description should have "Earth pony" instead of "earth pony". (That's the only description change I found.)

blackshadowshade commented:
> Pinkie Pie's description should have "Earth pony" instead of "earth pony". (That's the only description change I found.)

I've just fixed that and rebased.

TheOrgg commented:
Devious asked if I had any flavor text for Bunny Temple.

I came up with the following, though it is a bit long.

Brenda Lynn Temple, a.k.a. LdyShade, a.k.a. Bunny (to a select few) was brought to earth on the back of a rabbit and taken away astride a boar.  She was a fiery woman with a generous heart.  She liked the number five, and her favorite number was 25-- five fives.  Her favorite color was any version of blue, and she loved when you lost 'the game.'  Her favorite Cheapass Game was Kill Doctor Lucky.  She loved bunnies, skunks, and dragons.  Stars made her laugh with their incandescent gas. When playing Brawl, Button Men, or annoyed with someone, she liked to pretend to beat people up.

cgolubi1 commented:
TheOrgg, are you hoping we'll review and add the Bunny Temple flavor text as part of this PR, or that we'd do it separately later?

If the former, can someone volunteer to take a look at the draft text TheOrgg wrote in https://github.com/buttonmen-dev/buttonmen/pull/2981#issuecomment-2571734471 and give it a review?

TheOrgg commented:
I had assumed it'd be a part of this, as that seems simpler.

dwvanstone commented:
The text doesn't seem too long. I'd remove the double hyphen and use parentheses here:  "She liked the number five, and her favorite number was 25 (five fives.)"

Do we need to make clear that this and future buttons in Tributes are for people who passed away or is that clear from the title? (I.e., do we need something like "(1972-2021)" (made up dates) in the description?)

TheOrgg commented:
I would rather keep the em-dash.

As for whether it explicitly lists someone's range of life, I'd rather it be up to the individuals who write their farewells.  I hinted at her age in the text, but didn't make it explicit.  I don't think it should be a requirement.

blackshadowshade commented:
Flavour text added to the pull request, rebase done.

cgolubi1 commented:
I stood up a new dev site at https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com with the latest code.

TheOrgg commented:
When trying to check BT's flavor text I got the following error:  "Internal error when calling loadButtonData"

blackshadowshade commented:
I think this is due to an encoding issue with the em-dash. I've changed it to an unambiguous HTML encoding instead. Let's see if that fixes the issue.

cgolubi1 commented:
TheOrgg: just a reminder that James is pushing new code to this PR, but i'm pushing the code to the dev site where you can look at whether it works.  So the dev site hasn't changed, because i haven't redeployed it yet; we can't tell whether James's fix has worked or not.

James: looks like circleci failed on that latest commit.  (And i'd slightly recommend against rebasing while we're midconversation about debugging something, though obviously it's not a big deal.)

blackshadowshade commented:
Ah, that's better. Tests now pass. Hooray for tests that pick up my silly oversights!

cgolubi1 commented:
New dev site is finally up, and it looks like the encoding issues on https://2535-new-buttons.blackshadowshade.dev.buttonweavers.com/ui/buttons.html?button=Bunny%20Temple have been fixed - can y'all take a look?

TheOrgg commented:
Sometimes we'd tease her by setting a volume to 4 or 6... or 24, even.  That the button looks like it has 25, but the V isn't able to be a 5 would annoy her.  And those little annoyances were ways we showed we cared...

I don't think that's needed in the text, though.  I just wanted to share the reason for the recipe.

It looks good.

cgolubi1 commented:
TheOrgg, thanks for your work on these buttons.  Actually seeing the art and flavor text gave us the context to realise we have concerns around publishing tribute buttons for real people in general.  We're not comfortable making the call to tie someone's names and handles together publically, and we're not comfortable assuming that someone who wasn't part of the buttonweavers community per se would want to be memorialized by that community.

We think we need to restrict tribute buttons to active buttonweavers players and cases where we can collectively make the informed call both that the person would want to be referenced on buttonweavers and how they would want to be referenced.

Apologies that you had to be our test case on figuring this out, and that it took so long for us to get to a conclusion as a result.  :>(

TheOrgg commented:
Understood.

TheOrgg commented:
Are the rest good to go?

cgolubi1 commented:
Yes, the rest should be good to go.  Shadowshade is going to do a new revision as soon as he has a chance.

cgolubi1 commented:
I quickly stood up a new dev site and took a look, and this looks fine.  Merging now, thanks everyone!
